### ISSUE TITLE
`vitest/max-nested-describe` false positive

### ISSUE THREAD

**Issue by Maxim-Mazurok** [veritem/eslint-plugin-vitest#23]
For the following code it highlights the last 2 describe blocks:

```ts
import { CurrencyType } from "./constants.js";
import { Currency } from "./Currency.js";

describe("currency", () => {
  describe("constructor", () => {});

  describe("normalise currency name", () => {});

  describe("get canonical name and multiplier", () => {});

  describe("to string", () => {});
});
```

with this configuration:
```json
"vitest/max-nested-describe": [
          "error",
          {
            "max": 3
          }
        ],
```

If I increase max to 4 - it will only highlight the last block.
I think that nesting isn't counted correctly.

**Comment by veritem** [comment]
@Maxim-Mazurok thank you for reporting this. Nested block here means that describe blocks are inside other describe block like the following 

```js
describe('another suite', () => {
 describe('another suite', () => {
    describe('another suite', () => {
       describe('another suite', () => {
          describe('another suite', () => {
          describe('another suite', () => {
    
          })
       })
      })
    })
  })
})
```

**Comment by veritem** [comment]
would you like me to write another rule that covers your use case?

**Comment by Maxim-Mazurok** [comment]
@veritem no, this is not a feature request, it's a bug report :)

I understand that the rule is supposed to prevent nested describe blocks:
```js
describe("bla", () => {
  describe("bla", () => {
      describe("bla", () => {
	      describe("bla", () => {
	      })
      })
  })
})
```

The problem is that the rule is being triggered and reports the following code as an error when it shouldn't:
```
describe("bla", () => {
  describe("bla", () => {})
  describe("bla", () => {})
  describe("bla", () => {})
  describe("bla", () => {})
})
```

Hopefully, this clears it up :)

**Comment by Maxim-Mazurok** [comment]
Issue seems to be resolved now

![image](https://user-images.githubusercontent.com/7756211/234479358-177fd996-eb48-4931-a36a-10efeb7ad4fd.png)
